pkg/resources/materialized_view_acceptance_test.go

```diff
@@ -91,17 +201,94 @@ resource "snowflake_table" "test" {
 
 resource "snowflake_materialized_view" "test" {
 	name      = "%s"
-	comment   = "Terraform test resource"
+	comment   = "%s"
 	database  = "%s"
 	schema    = "%s"
 	warehouse = snowflake_warehouse.wh.name
-	is_secure = true
-	or_replace = false
+	is_secure = %t
+	or_replace = %t
 	statement = "%s"
 
 	depends_on = [
   		snowflake_table.test
   	]
 }
-`, warehouseName, tableName, databaseName, schemaName, viewName, databaseName, schemaName, q)
+`, warehouseName, tableName, databaseName, schemaName, viewName, comment, databaseName, schemaName, isSecure, orReplace, q)
+}
+
+func materializedViewConfigWithTags(warehouseName string, tableName string, viewName string, q string, databaseName string, schemaName string, tag string) string {
+	return fmt.Sprintf(`
+resource "snowflake_warehouse" "wh" {
+	name = "%s"
+}
+
+resource "snowflake_table" "test" {
+	name     = "%s"
+	database = "%s"
+	schema   = "%s"
+
+	column {
+		name = "ID"
+		type = "NUMBER(38,0)"
+	}
+}
+
+resource "snowflake_tag" "test_tag" {
+	name     = "tag1"
+	database = "%s"
+	schema   = "%s"
+}
+
+resource "snowflake_tag" "test_tag_2" {
+	name     = "tag2"
+	database = "%s"
+	schema   = "%s"
+}
+
+resource "snowflake_materialized_view" "test" {
+	name      = "%s"
+	database  = "%s"
+	schema    = "%s"
+	warehouse = snowflake_warehouse.wh.name
+	statement = "%s"
+
+	tag {
+		name = snowflake_tag.%s.name
+		schema = snowflake_tag.%s.schema
+		database = snowflake_tag.%s.database
+		value = "some_value"
+	}
+}
+`, warehouseName, tableName, databaseName, schemaName, databaseName, schemaName, databaseName, schemaName, viewName, databaseName, schemaName, q, tag, tag, tag)
+}
+
+func testAccCheckMaterializedViewDestroy(s *terraform.State) error {
+	db := acc.TestAccProvider.Meta().(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "snowflake_materialized_view" {
+			continue
+		}
+		ctx := context.Background()
+		id := sdk.NewSchemaObjectIdentifier(rs.Primary.Attributes["database"], rs.Primary.Attributes["schema"], rs.Primary.Attributes["name"])
+		existingMaterializedView, err := client.MaterializedViews.ShowByID(ctx, id)
+		if err == nil {
+			return fmt.Errorf("materialized view %v still exists", existingMaterializedView.ID().FullyQualifiedName())
+		}
+	}
+	return nil
+}
+
+func alterMaterializedViewQueryExternally(t *testing.T, id sdk.SchemaObjectIdentifier, query string, warehouse string) {
+	t.Helper()
+
+	client, err := sdk.NewDefaultClient()
+	require.NoError(t, err)
+	ctx := context.Background()
+
+	err = client.Sessions.UseWarehouse(ctx, sdk.NewAccountObjectIdentifier(warehouse))
```

Comment:
I tried without it and some tests failed without it sometimes. I left it as a discussion topic in SNOW-867235 because this is not the only resource behavinglike that.